contrib/cf-remote/cf_remote/remote.py

```diff
@@ -206,3 +225,25 @@ def install_host(
             demo_lib.agent_run(data, connection=connection)
             demo_lib.disable_password_dialog(host)
         demo_lib.agent_run(data, connection=connection)
+
+
+@auto_connect
+def uninstall_host(host, *, connection=None):
+    data = get_info(host, connection=connection)
+    print_info(data)
+
+    if not data["agent_version"]:
+        print("CFEngine is not installed on '{}' - moving on".format(host))
+        return data
+
+    uninstall_cfengine(host, data, connection=connection)
+    data = get_info(host, connection=connection)
+
+    if (not data) or data["agent_version"]:
```

Comment:
Does `not data` really mean uninstallation failed?

Reply:
I think so. We are unable to connect and verify that cfengine is there, so it's a failure.